pkg/prometheus/promcfg.go

```diff
@@ -2278,7 +2278,13 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	sc *monitoringv1alpha1.ScrapeConfig,
 	store *assets.Store,
 ) (yaml.MapSlice, error) {
-	jobName := fmt.Sprintf("scrapeConfig/%s/%s", sc.Namespace, sc.Name)
+	isJobNameSet := sc.Spec.JobName != ""
+	jobName := sc.Spec.JobName
+
+	if !isJobNameSet {
+		jobName = fmt.Sprintf("scrapeConfig/%s/%s", sc.Namespace, sc.Name)
+	}
```

Comment:
```suggestion
	jobName := fmt.Sprintf("scrapeConfig/%s/%s", sc.Namespace, sc.Name)
	if sc.Spec.JobName != nil {
		jobName = sc.Spec.JobName
	}
```

We can save one memory allocation while improving readability like this :)

Reply:
thanks! I reverted the changes due to using the relabel_config now